src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -1127,42 +1126,60 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
         }
       }
 
-      if (returnFiber.tag === HostRoot) {
-        // Top-level only accepts elements or portals
-        invariant(
-          // If the root has an error effect, this is an intentional unmount.
-          // Don't throw an error.
-          returnFiber.effectTag & Err,
-          'render(): Invalid component element.'
-        );
-      } else {
-        switch (returnFiber.tag) {
-          case ClassComponent: {
-            if (__DEV__) {
-              const instance = returnFiber.stateNode;
-              if (instance.render._isMockFunction) {
-                // We allow auto-mocks to proceed as if they're
-                // returning null.
-                break;
-              }
+      switch (returnFiber.tag) {
```

Comment:
Seems unnecessary to fork this whole function now. Can't you just add this conditional down below behind a smaller `if (ReactFeatureFlags.disableNewFiberFeatures) {` instead of forking the whole function?